src/txmempool.h

```diff
@@ -49,6 +49,20 @@ struct LockPoints
     LockPoints() : height(0), time(0), maxInputBlock(nullptr) { }
 };
 
+struct CompareIteratorByHashGeneric {
+    // SFINAE for T where T is either a pointer type (e.g., a txiter) or a reference_wrapper<T>
+    // (e.g. a wrapped CTxMemPoolEntry&)
+    template <typename T>
+    bool operator()(const std::reference_wrapper<T>& a, const std::reference_wrapper<T>& b) const
+    {
+        return a.get().GetTx().GetHash() < b.get().GetTx().GetHash();
+    }
+    template <typename T>
+    bool operator()(const T& a, const T& b) const
+    {
+        return a->GetTx().GetHash() < b->GetTx().GetHash();
+    }
+};
```

Comment:
nit: Could just call this `CompareIteratorByHash` given there's no alternative definition.

Reply:
Fair -- I think when i originally wrote the generic class, I didn't delete the prior class definition, but realized I could with the SFINAE at some point. 

If I have to make other changes, can adjust the name.

Reply:
adjusted.